### PR TITLE
Duplicate function and method inspections combined into one class

### PR DESCRIPTION
**Pull request by dlsniper**
Continuing on the work from @linniksa on https://github.com/go-lang-plugin-org/go-lang-idea-plugin/pull/1603

### CODE REVIEW

**Comment by dlsniper** [comment]
Done, tests should be updated now and all should work.

File: src/com/goide/inspections/GoDuplicateFunctionOrMethodInspection.java [hunk 1]
```diff
@@ -33,19 +36,56 @@
...
     Project project = file.getProject();
     String packageName = file.getPackageName();
     GlobalSearchScope scope = GoPsiImplUtil.packageScope(file);
 
+    for (final GoFunctionDeclaration func : file.getFunctions()) {
+      if (func.isBlank()) continue;
+
+      final String funcName = func.getName();
+      if (funcName == null) continue;
+      if (INIT.equals(funcName) & zeroArity(func)) continue;
```
Comment (zolotov) [inline src/com/goide/inspections/GoDuplicateFunctionOrMethodInspection.java]: & => &&

File: src/com/goide/inspections/GoDuplicateFunctionOrMethodInspection.java [hunk 2]
```diff
@@ -33,19 +36,56 @@
 
 import java.util.Collection;
 
-public class GoDuplicateMethodInspection extends GoInspectionBase {
+import static com.goide.GoConstants.INIT;
+import static com.goide.GoConstants.MAIN;
+
+public class GoDuplicateFunctionOrMethodInspection extends GoInspectionBase {
   @Override
   protected void checkFile(@NotNull GoFile file, @NotNull ProblemsHolder problemsHolder) {
```
Comment (zolotov) [inline src/com/goide/inspections/GoDuplicateFunctionOrMethodInspection.java]: Shouldn't be check file, build a visitor

**Comment by dlsniper** [comment]
Done. I've made a separate commit so that we still keep the original contribution.

**Pull request merged**